src/engine/GroupBy.cpp

```diff
@@ -828,8 +829,16 @@ std::optional<GroupBy::HashMapAggregateType> GroupBy::isSupportedAggregate(
   // `expr` is not a nested aggregated
   if (expr->children().front()->containsAggregate()) return std::nullopt;
 
-  if (hasType<AvgExpression>(expr)) return HashMapAggregateType::AVG;
-  if (hasType<CountExpression>(expr)) return HashMapAggregateType::COUNT;
+  if (hasType<AvgExpression>(expr)) return HashMapAggregateTypeWithData{AVG};
```

Comment:
if you sprinkle in a local `using H = HashMapAggregateTypeWithData` and use it,
then probably everything fits in one line.